libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -663,6 +666,38 @@ public void setTitlePlaceholder(CharSequence placeholderText) {
     public void setContentPlaceholder(CharSequence placeholderText) {
     }
 
+    // Getting the content from the HTML editor can take time and the UI seems to be unresponsive.
+    // Show a progress dialog for now. Ref: https://github.com/wordpress-mobile/gutenberg-mobile/issues/713
+    @Override
+    public boolean showSavingProgressDialogIfNeeded() {
+        if (!isAdded()) {
+            return false;
+        }
+
+        if (!mHtmlModeEnabled) return false;
+
+        if (mSavingContentProgressDialog != null && mSavingContentProgressDialog.isShowing()) {
+            // Already on the screen? no need to show it again.
+            return true;
+        }
+
+        mSavingContentProgressDialog = new ProgressDialog(getActivity());
```

Comment:
minor nit: can we check `mSavingContentProgressDialog == null` and only create a new ProgressDialog in such case? something like this:
```
        if (mSavingContentProgressDialog == null) {
            mSavingContentProgressDialog = new ProgressDialog(getActivity());
            mSavingContentProgressDialog.setCancelable(false);
            mSavingContentProgressDialog.setIndeterminate(true);
            mSavingContentProgressDialog.setMessage(getActivity().getString(R.string.long_post_dlg_saving));
        }
        mSavingContentProgressDialog.show();
```

Reply:
Thanks Mario, I've address your comment in https://github.com/wordpress-mobile/WordPress-Android/pull/9462/commits/2ca115b2d4f2c8a2426275622a5f4fe8572195bc, let wait the green flag before merging!